--- a/audio/out/ao_coreaudio.c
+++ b/audio/out/ao_coreaudio.c
@@ -139 +139 @@
-static bool init_audiounit(struct ao *ao, AudioStreamBasicDescription asbd, AudioChannelLayout *layout, size_t layout_size);
+static bool init_audiounit(struct ao *ao, AudioStreamBasicDescription asbd);
@@ -181,2 +180,0 @@
-    size_t layout_size;
-    AudioChannelLayout *layout = ca_get_acl(ao, &layout_size);
@@ -184,4 +182 @@
-    bool r = init_audiounit(ao, asbd, layout, layout_size);
-    talloc_free(layout);
-
-    if (!r)
+    if (!init_audiounit(ao, asbd))
@@ -280 +275 @@
-static bool init_audiounit(struct ao *ao, AudioStreamBasicDescription asbd, AudioChannelLayout *layout, size_t layout_size)
+static bool init_audiounit(struct ao *ao, AudioStreamBasicDescription asbd)
@@ -323,7 +317,0 @@
-
-    err = AudioUnitSetProperty(p->audio_unit,
-                               kAudioOutputUnitProperty_ChannelMap,
-                               kAudioUnitScope_Global, 0, layout, layout_size);
-
-    CHECK_CA_ERROR_L(coreaudio_error_audiounit,
-                     "unable to set the input channel layout on the audio unit");
